### PR TITLE
Add core plugins snapshots publishing alongside min snapshots

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Add core plugins snapshots publishing alongside min snapshots

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/5096

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -128,7 +128,7 @@ jacocoTestReport {
     }
 }
 
-String version = '7.3.1'
+String version = '7.3.2'
```
Comment (prudhvigodithi) [inline build.gradle:131]: Hey @peterzhuamazon you can remove this as I have change pushed which has `7.3.2`.
https://github.com/opensearch-project/opensearch-build-libraries/pull/559
Thanks

Comment (peterzhuamazon) [reply build.gradle:131]: Thanks 👍

**Review by prudhvigodithi — APPROVED** [review]

File: vars/uploadMinSnapshotsToS3.groovy [hunk 2]
```diff
@@ -19,6 +19,10 @@ void call(Map args = [:]) {
 
     def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: manifest))
     String productName = inputManifest.build.getFilename()
+    String version_plain = inputManifest.build.version
+    String qualifier = inputManifest.build.qualifier ? '-' + inputManifest.build.qualifier : ''
+    String revision = version_plain + qualifier + '-SNAPSHOT'
```
Comment (gaiksaya) [inline vars/uploadMinSnapshotsToS3.groovy:24]: We build with snapshot=true. https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch/publish-min-snapshots.jenkinsfile#L73
Do we still need revision here?

File: vars/uploadMinSnapshotsToS3.groovy [hunk 3]
```diff
@@ -68,9 +83,29 @@ void call(Map args = [:]) {
...
             withAWS(role: "${ARTIFACT_PROMOTION_ROLE_NAME}", roleAccount: "${AWS_ACCOUNT_ARTIFACT}", duration: 900, roleSessionName: 'jenkins-session') {
+                // min artifacts
+                echo("Upload min snapshots")
                 s3Upload(file: "${srcDir}/${baseName}-latest.${extension}", bucket: "${ARTIFACT_PRODUCTION_BUCKET_NAME}", path: "${dstDir}/${baseName}-latest.${extension}")
                 s3Upload(file: "${srcDir}/${baseName}-latest.${extension}.sha512", bucket: "${ARTIFACT_PRODUCTION_BUCKET_NAME}", path: "${dstDir}/${baseName}-latest.${extension}.sha512")
                 s3Upload(file: "${srcDir}/${baseName}-latest.${extension}.build-manifest.yml", bucket: "${ARTIFACT_PRODUCTION_BUCKET_NAME}", path: "${dstDir}/${baseName}-latest.${extension}.build-manifest.yml")
+                // core plugins
+                if (architecture == "x64" && platform == "linux" && distribution == "tar" && corePluginDirExists) {
+                    echo("Upload core-plugins snapshots")
+                    List<String> corePluginList = buildManifestYamlOnly.components.artifacts.'core-plugins'[0]
```
Comment (gaiksaya) [inline vars/uploadMinSnapshotsToS3.groovy:94]: Why not retrieve component from already existing object `buildManifest`. We do have component class in it https://github.com/opensearch-project/opensearch-build-libraries/blob/main/src/jenkins/BuildManifest.groovy#L64

File: vars/uploadMinSnapshotsToS3.groovy (hunk 2 above)
Comment (peterzhuamazon) [reply vars/uploadMinSnapshotsToS3.groovy:24]: We do, revision is more for the `qualifier`, such as `3.0.0-beta1`, `3.0.0-rc1`.

File: vars/uploadMinSnapshotsToS3.groovy (hunk 3 above)
Comment (peterzhuamazon) [reply vars/uploadMinSnapshotsToS3.groovy:94]: This is a exact copy paste from promote artifact, we can improve these both later on.
Thanks.

File: vars/uploadMinSnapshotsToS3.groovy [hunk 4]
```diff
@@ -68,9 +83,29 @@ void call(Map args = [:]) {
         string(credentialsId: 'jenkins-aws-production-account', variable: 'AWS_ACCOUNT_ARTIFACT'),
         string(credentialsId: 'jenkins-artifact-production-bucket-name', variable: 'ARTIFACT_PRODUCTION_BUCKET_NAME')]) {
             withAWS(role: "${ARTIFACT_PROMOTION_ROLE_NAME}", roleAccount: "${AWS_ACCOUNT_ARTIFACT}", duration: 900, roleSessionName: 'jenkins-session') {
+                // min artifacts
+                echo("Upload min snapshots")
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply vars/uploadMinSnapshotsToS3.groovy:87]: Will keep it for now as it is grammar.

**Pull request merged**